util/tls/tls_socket.h

```diff
@@ -43,6 +43,13 @@ class TlsSocket : public FiberSocketBase {
 
   SSL* ssl_handle();
 
+  // Note:
+  // This is not part of the FiberSocketBase.
+  // This function is mostly useful for client side,
+  // since part of the TLS is to handshake after the connection
+  // is done using "normal" TCP connection.
+  error_code Handshake(Engine::HandshakeType Handshake_type);
```

Comment:
who do you need to expose this method? why not put this functionality inside TlsSocket::Connect method that is currently empty?

Reply:
I tried put it here (it was there originally, don't remember why I remove it from there, I will try to put it back here again, maybe because of other issues I had I though I should move it out